tests/schemas/test_openapi.py

```diff
@@ -51,6 +51,10 @@ def test_list_field_mapping(self):
         uuid1 = uuid.uuid4()
         uuid2 = uuid.uuid4()
         inspector = AutoSchema()
+
+        class NestedSerializer(serializers.Serializer):
+            data = serializers.ListSerializer(child=serializers.FloatField())
+
```

Comment:
This looks like a usage error. 

`ListSerializer` is for many **objects**, not scalars, for which `ListField` is used. 

Normally you get a `ListSerializer` using a nested serializer with `many=True`. 

```
class CommentSerializer(serializers.Serializer):
    user = UserSerializer(required=False)
    edits = EditItemSerializer(many=True)  # A nested list of 'edit' items.
```

https://www.django-rest-framework.org/api-guide/serializers/#dealing-with-nested-objects
https://www.django-rest-framework.org/api-guide/serializers/#listserializer

Do we have a test for the `many=True` case? 👀  Would that still pass with this change?

Reply:
Ahh...

Okey, my bad.

But is it of interest to include an better error message? I'm thinking that if a ListSerializer with a field, it raises an exception proposing to use a ListField instead?

Reply:
Yep... that kind of thing would be good look at. 

Thanks!